clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java

```diff
@@ -60,6 +61,51 @@ public void partitionAssignment() {
     }
 
     @Test
+    public void partitionAssignmentOnTopicSubscription() {
+        state.assignFromUser(new HashSet<>(Arrays.asList(tp0, tp1)));
+        // assigned partitions should immediately change
+        assertEquals(2, state.assignedPartitions().size());
+        assertTrue(state.assignedPartitions().contains(tp0));
+        assertTrue(state.assignedPartitions().contains(tp1));
+
+        state.unsubscribe();
+        // assigned partitions should immediately change
+        assertTrue(state.assignedPartitions().isEmpty());
+
+
+        state.subscribe(singleton(topic1), rebalanceListener);
+        // assigned partitions should remain unchanged
+        assertTrue(state.assignedPartitions().isEmpty());
```

Comment:
This doesn't seem too interesting as a test case since we already unsubscribed. Maybe the second `subscribe` can go before the `unsubscribe`? Then we can verify that we still own tp0 and tp1.